src/TypeSpec.Extension/Emitter.Csharp/src/lib/operation.ts

```diff
@@ -380,7 +383,8 @@ export function loadOperation(
         op: HttpOperation,
         resourceOperation?: ResourceOperation
     ): OperationLongRunning | undefined {
-        if (!isLongRunningOperation(context, op.operation)) return undefined;
+        let metadata = getLroMetadata(program, op.operation);
+        if (metadata === undefined) return undefined;
 
         const finalResponse = loadLongRunningFinalResponse(
```

Comment:
The reason I'm not using the result model is because we have incoming swagger definition as well. Our implementation of LRO in the generator depends on the response, not the model, so I'm keeping the current behavior unless there is strong reason to re-write the whole LRO generation logic.